envoy/http/early_header_mutation.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/http/header_map.h"
+#include "envoy/server/factory_context.h"
+#include "envoy/stream_info/stream_info.h"
+
+namespace Envoy {
+namespace Http {
+
+/**
+ * Interface class for early header mutation extensions.
+ */
+class EarlyHeaderMutation {
+public:
+  virtual ~EarlyHeaderMutation() = default;
+
+  /**
+   * Mutate the request headers at very early.
```

Comment:
nit: Not very clear. Can you be more clear about when this happens?

Reply:
right. got it.

Reply:
comment updated.